FizzBuzz/src/main/java/rocks/kata/FizzBuzzDivisibilityRules.java

```diff
@@ -0,0 +1,65 @@
+package rocks.kata;
+
+import java.util.Map;
+import static rocks.kata.DivisibilityStatus.*;
+
+public class FizzBuzzDivisibilityRules implements FizzBuzz {
+
+    private Map<DivisibilityStatus, String> fizzBuzzTranslation = Map.of(
+            DIVISIBLE_BY_3, "Fizz",
+            DIVISIBLE_BY_5, "Buzz",
+            DIVISIBLE_BY_15, "FizzBuzz");
+
+    @Override
+    public String returnFizzBuzzOrNumber(int number) {
+        DivisibilityStatus checkDivisibilityStatus = checkDivisibility(number);
+
+        return this.fizzBuzzTranslation.getOrDefault(checkDivisibilityStatus, Integer.toString(number));
```

Comment:
W przypadkach kiedy nie ma wątpliwości, że dana nazwa odnosi się do pola (poza konstruktorem i setterami zwykle wszędzie), najlepiej pomijać `this`, bo nic nie wnosi, a zmniejsza czytelność.